backend/src/services/integrationService.ts

```diff
@@ -170,7 +233,21 @@ export default class IntegrationService {
         for (const id of ids) {
           let integration
           try {
+            const collectionService = new CollectionService(this.options)
             integration = await this.findById(id)
+
+            if (integration?.segmentId) {
+              const [insightsProject] = await collectionService.findInsightsProjectsBySegmentId(
+                integration.segmentId,
+              )
+
+              if (insightsProject) {
+                await collectionService.updateInsightsProject(insightsProject.id, {
+                  repositories: [],
+                  widgets: [],
+                })
```

Comment:
We should add some validations here. We should only completely clean repositories and widgets if ALL segments integrations were destroyed.

Since this iterates through an array of destroyed integrations we should probably check by the end of this processing which platforms were destroyed, and for those we remove the repositories for the specific platforms, and the widgets for the specific platforms. Something like:
1. Delete integrations from the ids input
2. Check which integration platforms were deleted
3. For the repositories linked to the deleted platform, remove them from insightsProjects.repositoreis
4. For the widgets linked to the deleted platform, remove them from insightsProjects.widgets